dali/operators/decoder/video/video_decoder_cpu.cc

```diff
@@ -12,66 +12,143 @@
+        encoded=encoded_video,
+        start_frame=0, sequence_length=10, stride=2
+        ...,
+    )
+
+Example 3: Pad the sequence with the last frame in the video, until 100 frames are reached:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="edge"
+        ...,
+    )
+
+Example 4: Pad the sequence with a constant value of 128, until 100 frames are reached:
 
-The video streams can be in most of the container file formats. FFmpeg is used to parse video
- containers and returns a batch of sequences of frames with shape (F, H, W, C) where F is the
- number of frames in a sequence and can differ for each sample.)code")
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="constant", fill_value=128
+        ...,
+
+Example 5: Pad the sequence with a constant RGB value of (118, 185, 0), until 100 frames are reached:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="constant", fill_value=[118, 185, 0]
+        ...,
+)code")
     .NumInput(1)
     .NumOutput(1)
-    .InputDox(0, "buffer", "TensorList", "Data buffer with a loaded video file.")
+    .InputDox(0, "encoded", "TensorList", "Encoded video stream")
     .AddOptionalArg("affine",
-    R"code(Applies only to the mixed backend type.
+                    R"code(Whether to pin threads to CPU cores (mixed backend only).
+
+If True, each thread in the internal thread pool will be pinned to a specific CPU core.
+If False, threads can migrate between cores based on OS scheduling.)code",
+                    true)
+    .AddOptionalArg<std::vector<int>>(
+        "frames",
+        R"code(Specifies which frames to extract from each video by their indices.
+
+The indices can be provided in any order and can include duplicates. For example, ``[0,10,5,10]`` would extract:
+
+* Frame 0 (first frame)
+* Frame 10 
+* Frame 5
+* Frame 10 (again)
+
+This argument cannot be used together with ``start_frame``, ``sequence_length``, ``stride``.)code",
+        nullptr, true)
+    .AddOptionalArg<int>(
+        "start_frame",
+        R"code(Index of the first frame to extract from each video. Cannot be used together with ``frames`` argument.)code",
+        nullptr, true)
+    .AddOptionalArg<int>(
+        "stride",
+        R"code(Number of frames to skip between each extracted frame. Cannot be used together with ``frames`` argument.)code",
+        nullptr, true)
+    .AddOptionalArg<int>(
+        "sequence_length",
+        R"code(Number of frames to extract from each video. If not provided, the whole video is decoded. Cannot be used together with ``frames`` argument.)code",
```

Comment:
If I read this correctly, this implies that there will be only one sequence per video file? Is that correct?